src/shared/pretty-print.c

```diff
@@ -531,28 +531,24 @@ void clear_progress_bar_unbuffered(const char *prefix) {
 }
 
 void draw_progress_bar(const char *prefix, double percentage) {
-
         /* We are going output a bunch of small strings that shall appear as a single line to STDERR which is
          * unbuffered by default. Let's temporarily turn on full buffering, so that this is passed to the tty
          * as a single buffer, to make things more efficient. */
-        char buffer[LONG_LINE_MAX];
-        setvbuf(stderr, buffer, _IOFBF, sizeof(buffer));
-
+        WITH_BUFFERING(stderr);
         draw_progress_bar_unbuffered(prefix, percentage);
-
-        fflush(stderr);
-
-        /* Disable buffering again */
-        setvbuf(stderr, NULL, _IONBF, 0);
 }
 
 void clear_progress_bar(const char *prefix) {
-        char buffer[LONG_LINE_MAX];
-        setvbuf(stderr, buffer, _IOFBF, sizeof(buffer));
-
+        WITH_BUFFERING(stderr);
         clear_progress_bar_unbuffered(prefix);
+}
 
-        fflush(stderr);
+FILE* flush_and_disable_buffering(FILE *p) {
+        if (!p)
+                return NULL;
 
-        setvbuf(stderr, NULL, _IONBF, 0);
+        fflush(p);
+        /* Disable buffering again */
+        setvbuf(p, NULL, _IONBF, 0);
+        return NULL;
```

Comment:
The return value is not used. Maybe, just introduce a void func that takes FILE** directly?